Modules/Core/Common/include/itkLeafTreeIterator.h

```diff
@@ -134,6 +134,10 @@ const typename LeafTreeIterator<TTreeType>::ValueType &
 LeafTreeIterator<TTreeType>::Next()
 {
   this->m_Position = const_cast<TreeNodeType *>(FindNextNode());
+  if (this->m_Position == nullptr)
+  {
+    return this->m_Root->Get(); // value irrelevant, but we have to return something
```

Comment:
Sorry, I'm not familiar with this class. Is it still being used? If so, could a user application actually get at this line of code, during program execution? Should it get an assert failure here?

Reply:
A test gets there: https://open.cdash.org/testDetails.php?test=817215405&build=6229026

Reply:
I am not familiar with this class either, and I don't know if it is being used.

The unit test gets to here, but the result is commonly discarded, so the whole operation is commonly optimized away. But as Sean's tool notes, it is undefined behavior.

Reply:
The tool in question is -fsanitize=null https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html

Reply:
If the return value is commonly ignored, maybe just make it `void`. Otherwise I think an assert failure might be helpful. But again, I don't really know.

Reply:
`operator++` (iterator increment) returns this reference, which is common in C++, but the [usual use pattern](https://github.com/InsightSoftwareConsortium/ITK/blob/2074c2f9e5ed3f087d0e2059f8e0e8992fcad7ef/Modules/Core/Common/test/itkTreeContainerTest.cxx#L160-L167) does not use that reference. And since this group of classes seems to be used rarely (if at all), refactoring them is not a priority.